src/main/java/com/guysfromusa/carsgame/validator/CarNotInGameValidator.java

```diff
@@ -0,0 +1,39 @@
+package com.guysfromusa.carsgame.validator;
+
+import com.guysfromusa.carsgame.game_state.ActiveGamesContainer;
+import com.guysfromusa.carsgame.game_state.dtos.GameState;
+import com.guysfromusa.carsgame.v1.model.Car;
+import com.guysfromusa.carsgame.validator.subject.CarGameAdditionValidationSubject;
+import org.springframework.stereotype.Component;
+
+import javax.inject.Inject;
+import java.util.List;
+
+import static com.guysfromusa.carsgame.utils.StreamUtils.convert;
+import static org.apache.commons.lang3.Validate.notNull;
+
+/**
+ * Created by Sebastian Mikucki, 04.05.18
+ */
+@Component
+public class CarNotInGameValidator implements BusinessValidator<CarGameAdditionValidationSubject> {
+
+    public static final String CAR_EXISTS_IN_GAME_MESSAGE = "Car is already added to game";
+
+    private final ActiveGamesContainer activeGamesContainer;
+
+    @Inject
+    public CarNotInGameValidator(ActiveGamesContainer activeGamesContainer) {
+        this.activeGamesContainer = notNull(activeGamesContainer);
+    }
+
+    @Override
+    public void validate(CarGameAdditionValidationSubject subject) {
+        GameState gameState = activeGamesContainer.getGameState(subject.getGameEntity().getName());
+        List<String> gamesCarNames = convert(gameState.getAllCars(), Car::getName);
```

Comment:
you dont have to collect to list all cars to check if a car exist

stream.map.filter.findany.orelsethrow